prime-router/docker-compose.build.yml

```diff
@@ -0,0 +1,45 @@
+# Build environment
+version: "3.3"
+services:
+  builder:
+    depends_on:
+      - postgresql
+    build:
+      context: .
+      dockerfile: Dockerfile.build
+    environment:
+      FUNCTIONS_CORE_TOOLS_TELEMETRY_OPTOUT: 1
+      DB_URL: jdbc:postgresql://postgresql:5432/prime_data_hub
+    volumes: # Attach the pwd into the image
+      - type: bind
+        source: ./
+        target: /src
+    working_dir: /src
+    dns:
+      - 1.1.1.1
```

Comment:
Any reason for specifying the DNS?

Reply:
Not really, this is residual to my own personal setup, this will be removed when the PR exists draft status :) I'll add a TODO so that any review can more easily pick it up
good catch :)

Reply:
I had a bit of a think about the pre-built images and I think it makes sense to do that. We can tag them accordingly so that we have the ability to quickly flip between other combinations of JDK/Gradle.